File: services/pix-boleto-kotlin/src/main/kotlin/com/acme/payments/pixboleto/api/IdempotencyConflictException.kt
package com.acme.payments.pixboleto.api

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

@ResponseStatus(HttpStatus.CONFLICT)
class IdempotencyConflictException(key: String) :
    RuntimeException("idempotency key was already used with a different request: $key")

File: services/card-payment-csharp/src/CardPaymentService/Application/IdempotencyConflictException.cs
namespace Acme.Payments.CardPaymentService.Application;

public sealed class IdempotencyConflictException : InvalidOperationException
{
    public IdempotencyConflictException(string key)
        : base($"idempotency key was already used with a different request: {key}")
    {
    }
}
